confuse/cache.py

```diff
@@ -0,0 +1,121 @@
+from typing import Dict, List
+
+from . import templates
+from .core import ROOT_NAME, Configuration, ConfigView, RootView, Subview
+
+
+class CachedHandle(object):
+    """Handle for a cached value computed by applying a template on the view.
+    """
+    # some sentinel objects
+    _INVALID = object()
+    _MISSING = object()
+
+    def __init__(self, view: ConfigView, template=templates.REQUIRED) -> None:
+        self.value = self._INVALID
+        self.view = view
+        self.template = template
+
+    def get(self):
+        """Retreive the cached value from the handle.
+
+        Will re-compute the value using `view.get(template)` if it has been
+        invalidated.
+
+        May raise a `NotFoundError` if the underlying view has been
+        invalidated.
+        """
+        if self.value is self._MISSING:
+            # will raise a NotFoundError if no default value was provided
+            self.value = templates.as_template(self.template).get_default_value()
```

Comment:
With the caveat that I may have this all wrong: I _think_ it may be equivalent to just do `self.view.get(self.template)` in this case as well? That is, the logic in `get` should (I think!!) use the default value anyway, when the underlying data is missing.

_If_ that's the case, we can probably eliminate the distinction between "missing" and "invalid"… invalidated handles would automatically fall back to the default value by piggy-backing on that logic in plain old `get`.

Reply:
Removed the `missing` case completely.